src/panels/lovelace/cards/hui-sensor-card.ts

```diff
@@ -202,304 +43,35 @@ class HuiSensorCard extends LitElement implements LovelaceCard {
     return { type: "sensor", entity: foundEntities[0] || "", graph: "line" };
   }
 
-  @property() public hass?: HomeAssistant;
-
-  @property() private _config?: SensorCardConfig;
-
-  @property() private _history?: any;
-
-  private _date?: Date;
-
   public setConfig(config: SensorCardConfig): void {
     if (!config.entity || config.entity.split(".")[0] !== "sensor") {
       throw new Error("Specify an entity from within the sensor domain.");
     }
 
-    const cardConfig = {
-      detail: 1,
-      theme: "default",
-      hours_to_show: 24,
-      ...config,
+    let entityCardConfig: EntityCardConfig = {
+      type: "entity",
+      entity: config.entity,
+      theme: config.theme || "default",
+      ...(config.name && { name: config.name }),
+      ...(config.unit && { unit: config.unit }),
+      ...(config.icon && { icon: config.icon }),
```

Comment:
Can't we do:
```suggestion
    let entityCardConfig: EntityCardConfig = {
      ...config,
      type: "entity",
```

Reply:
We can sure. But then we are sending in extra config variables

Graph, detail, hours to show

Reply:
So pull those out for the footer first and then send the rest to the `entityCard`